File: KvizHubBackend/KvizHubBack/Models/Question.cs
namespace KvizHubBack.Models
{
    public enum QuestionType
    {
        SingleChoice,
        MultipleChoice,
        TrueFalse,
        FillIn
    }
    public class Question
    {
        public int Id { get; set; }
        public int QuizId { get; set; }
        public string Text { get; set; } = null!;
        public QuestionType Type { get; set; }
        public int Points { get; set; }
        public Quiz? Quiz { get; set; }
        public ICollection<Answer>? Answers { get; set; }
        public ICollection<UserAnswer>? UserAnswers { get; set; }
    }
}

File: KvizHubFrontend/kvizhub-frontend/src/models/Question.ts
import { AnswerDto } from "./Answer";

export interface QuestionUpdateDto {
  id: number;
  text: string;
  type: "SingleChoice" | "MultipleChoice" | "FillIn";
  points: number;
  quizId: number;
}

export interface QuestionDto {
  id: number;
  text: string;
  type: string;
  points: number;
  answers?: AnswerDto[];
}